### PR TITLE
Feature/rangefinder vl53l1x

### PR DESCRIPTION
**Pull request by nathanjel**
I have ported the STM VL53L1X ULD API (STSW-IMG009) and tested on MATEK F722PX board successfully. I hope this is useful. I know VL53L0X is supported, but I had only the VL53L1X in my drawer.

### CODE REVIEW

File: src/main/drivers/rangefinder/rangefinder_vl53l1x.c [hunk 1]
```diff
@@ -0,0 +1,1693 @@
...
+    uint16_t Distance;  /*!< ResultDistance */
+    uint16_t Ambient;   /*!< ResultAmbient */
+    uint16_t SigPerSPAD;/*!< ResultSignalPerSPAD */
+    uint16_t NumSPADs;  /*!< ResultNumSPADs */
+} VL53L1X_Result_t;
+
+/**
+ * @brief This function returns the SW driver version
+ */
+VL53L1X_ERROR VL53L1X_GetSWVersion(VL53L1X_Version_t *pVersion);
```
Comment (digitalentity) [inline src/main/drivers/rangefinder/rangefinder_vl53l1x.c]: This is dead code, called from nowhere.

File: src/main/drivers/rangefinder/rangefinder_vl53l1x.c [hunk 2]
```diff
@@ -0,0 +1,1693 @@
...
+
+/**
+ * @brief This function sets the sensor I2C address used in case multiple devices application, default address 0x52
+ */
+VL53L1X_ERROR VL53L1X_SetI2CAddress(busDevice_t * dev, uint8_t new_address);
```
Comment (digitalentity) [inline src/main/drivers/rangefinder/rangefinder_vl53l1x.c]: Dead code

File: src/main/drivers/rangefinder/rangefinder_vl53l1x.c [hunk 3]
```diff
@@ -0,0 +1,1693 @@
...
+    calXtalk = (uint16_t)(512*(AverageSignalRate*(1-(AverageDistance/TargetDistInMm)))/AverageSpadNb);
+    *xtalk = (uint16_t)((calXtalk*1000)>>9);
+    status = VL53L1_WrWord(dev, 0x0016, (uint16_t)calXtalk);
+    return status;
+}
+
+static void vl53l1x_Init(rangefinderDev_t * rangefinder)
+{
+    isInitialized = false;
+    VL53L1X_SensorInit(rangefinder->busDev);
```
Comment (digitalentity) [inline src/main/drivers/rangefinder/rangefinder_vl53l1x.c]: Why don't we check thet `VL53L1X_SensorInit` returned no error?

File: src/main/drivers/rangefinder/rangefinder_vl53l1x.c [hunk 4]
```diff
@@ -0,0 +1,1693 @@
...
+ * @brief This function clears the interrupt, to be called after a ranging data reading
+ * to arm the interrupt for the next data ready event.
+ */
+VL53L1X_ERROR VL53L1X_ClearInterrupt(busDevice_t * dev);
+
+/**
+ * @brief This function programs the interrupt polarity\n
+ * 1=active high (default), 0=active low
+ */
+VL53L1X_ERROR VL53L1X_SetInterruptPolarity(busDevice_t * dev, uint8_t IntPol);
```
Comment (digitalentity) [inline src/main/drivers/rangefinder/rangefinder_vl53l1x.c]: Dead code

**Review by digitalentity — CHANGES_REQUESTED** [review]
I understand that this is a direct and trivial port of ST's library, but I would still suggest refactoring the code to at least the following:
 - Remove functions that are never called
 - Make all functions that are not intended to be called from outside this file `static` to make it easier for the linker

**Comment by nathanjel** [comment]
> I understand that this is a direct and trivial port of ST's library, but I would still suggest refactoring the code to at least the following:
> 
> * Remove functions that are never called
> * Make all functions that are not intended to be called from outside this file `static` to make it easier for the linker

Hi, thanks for reply. I agree and will update the commit.

File: src/main/drivers/rangefinder/rangefinder_vl53l1x.c (hunk 1 above)
Comment (nathanjel) [reply src/main/drivers/rangefinder/rangefinder_vl53l1x.c]: This is addressed in most recent commit

File: src/main/drivers/rangefinder/rangefinder_vl53l1x.c (hunk 2 above)
Comment (nathanjel) [reply src/main/drivers/rangefinder/rangefinder_vl53l1x.c]: This is addressed in most recent commit

Comment (nathanjel) [reply src/main/drivers/rangefinder/rangefinder_vl53l1x.c]: This is addressed in most recent commit

File: src/main/drivers/rangefinder/rangefinder_vl53l1x.c (hunk 3 above)
Comment (nathanjel) [reply src/main/drivers/rangefinder/rangefinder_vl53l1x.c]: This is addressed in most recent commit

File: src/main/drivers/rangefinder/rangefinder_vl53l1x.c (hunk 4 above)
Comment (nathanjel) [reply src/main/drivers/rangefinder/rangefinder_vl53l1x.c]: This is addressed in most recent commit

**Review by digitalentity — APPROVED** [review]
LGTM

**Pull request merged**